cpp/src/arrow/compute/api_scalar.h

```diff
@@ -30,6 +30,10 @@
 #include "arrow/util/macros.h"
 #include "arrow/util/visibility.h"
 
+#ifdef ARROW_WITH_UTF8PROC
+#include <utf8proc.h>
```

Comment:
We shouldn't include utf8proc in public headers, IMHO. This is only necessary for the kernel implementation.